src/extensions/step_handlers/script_handler/src/script_handler.cpp

```diff
@@ -552,14 +578,32 @@ static ADUC_Result ScriptHandler_PerformAction(const std::string& action, const
 
     aduShellArgs.emplace_back(adushconst::target_data_opt);
     aduShellArgs.emplace_back(scriptFilePath);
+    commandLineArgs.emplace_back(scriptFilePath);
 
-    aduShellArgs.emplace_back(adushconst::target_options_opt);
-    aduShellArgs.emplace_back(action.c_str());
+    // Prepare arguments based on specified api version.
+    if (apiVer == nullptr || strcmp(apiVer, "2.0") == 0)
+    {
+        std::string backcompatAction = "--action-" + action;
+        aduShellArgs.emplace_back(adushconst::target_options_opt);
+        aduShellArgs.emplace_back(backcompatAction.c_str());
+        commandLineArgs.emplace_back(backcompatAction.c_str());
+    }
+    else if (strcmp(apiVer, "2.1") == 0)
```

Comment:
careful - string compares on versions are problematic.

If the user puts "2.01" is that "2.1"?  What about "2.10" is that "2.1"?  I think an integer would be better.

Reply:
I think this is ok. It's quite common to use "major.minor" without any leading zero.
For next iteration, if there's a breaking change, we could release another handler sxs that associates with the new version of update handler type. e.g. "microsoft/scripthandler:**x**"